### PR TITLE
[SC-49206] Add run and stop commands to the cli

### PR DESCRIPTION
**Pull request by calving-db**
Changes:
Adds run/stop commands which underneath just call the REST API.

Tested:
Ran `tox` tests.

### CODE REVIEW

**Review by mukulmurthy — COMMENTED** [review]
We're removing the ability to pass anything but a --pipeline-id to non-deploy commands (see https://github.com/databricks/databricks-cli/pull/317). Guessing that one will go in first, so might have to hold off on this for a second and build on top of that.

File: databricks_cli/pipelines/cli.py [hunk 1]
```diff
@@ -187,6 +187,49 @@ def reset_cli(api_client, pipeline_id):
...
+    """
+    Starts the execution of a delta pipelines run by starting the cluster and processing data.
+
+    Usage:
+
+    databricks pipelines run --pipeline-id 1234
+    """
+    _validate_pipeline_id(pipeline_id)
+        PipelinesApi(api_client).run(pipeline_id)
+    click.echo("Run triggered for pipeline {}".format(pipeline_id))
```
Comment (mukulmurthy) [inline databricks_cli/pipelines/cli.py:208]: Fix the indenting here?

I don't know Python well enough, but I thought this was an error.

File: databricks_cli/pipelines/cli.py [hunk 2]
```diff
@@ -187,6 +187,49 @@ def reset_cli(api_client, pipeline_id):
...
+    Stops the execution of a delta pipelines run by terminating the cluster. Processing of data can be resumed by
+    calling `run`.
+
+    Usage:
+
+    databricks pipelines stop --pipeline-id 1234
+    """
+    _validate_pipeline_id(pipeline_id)
+    PipelinesApi(api_client).stop(pipeline_id)
+    click.echo("Stop triggered for pipeline {}".format(pipeline_id))
```
Comment (mukulmurthy) [inline databricks_cli/pipelines/cli.py:230]: Nit: "Stopped pipeline"

File: databricks_cli/pipelines/api.py [hunk 3]
```diff
@@ -63,6 +63,12 @@ def get(self, pipeline_id, headers=None):
     def reset(self, pipeline_id, headers=None):
         self.client.reset(pipeline_id, headers)
 
+    def run(self, pipeline_id, headers=None):
+        self.client.run(pipeline_id, headers)
+
+    def stop(self, pipeline_id, headers=None):
```
Comment (mukulmurthy) [inline databricks_cli/pipelines/api.py:69]: I know this is pretty basic, but we should still add tests for it in test_api.py.

File: databricks_cli/sdk/service.py [hunk 4]
```diff
@@ -938,3 +938,13 @@ def reset(self, pipeline_id=None, headers=None):
         _data = {}
 
         return self.client.perform_query('POST', '/pipelines/{pipeline_id}/reset'.format(pipeline_id=pipeline_id), data=_data, headers=headers)
+
+    def stop(self, pipeline_id=None, headers=None):
+        _data = {}
+
+        return self.client.perform_query('POST', '/pipelines/{pipeline_id}/stop'.format(pipeline_id=pipeline_id), data=_data, headers=headers)
+
+    def run(self, pipeline_id=None, headers=None):
```
Comment (mukulmurthy) [inline databricks_cli/sdk/service.py:947]: Are these generated from the protos or did you do it manually?

Comment (calving-db) [reply databricks_cli/sdk/service.py:947]: Oh, makes sense why the code looks so weird. Didn't realize they were generated. Do you happen to know the command to do that?

File: databricks_cli/pipelines/cli.py (hunk 1 above)
Comment (calving-db) [reply databricks_cli/pipelines/cli.py:208]: Yeah that is strange, all the tests passed though. Going to look into it.

File: databricks_cli/sdk/service.py (hunk 4 above)
Comment (mukulmurthy) [reply databricks_cli/sdk/service.py:947]: `bb //api:doc_and_sdk`. In theory you just copy the service.py file that gets generated, but they're out of sync, so just grab the pipelines bits that get generated. I think it's probably going to be exactly what you have.

File: databricks_cli/pipelines/cli.py (hunk 1 above)
Comment (mukulmurthy) [reply databricks_cli/pipelines/cli.py:208]: Actually it looks like the build on this PR did fail

Comment (calving-db) [reply databricks_cli/pipelines/cli.py:208]: Yeah you're right, I think i just messed up when running tox then

File: databricks_cli/sdk/service.py (hunk 4 above)
Comment (calving-db) [reply databricks_cli/sdk/service.py:947]: Nice, will confirm that it's the same

**Comment by calving-db** [comment]
Sorry for the million commit emails, something between virtualenv/tox was really not cooperating on my machine and I didn't anticipate having so many bugs lol

**Review by mukulmurthy — APPROVED** [review]

**Pull request merged**